### DISCUSSION TITLE
Performance is not improved for too complex query

### DISCUSSION THREAD

**Discussion by njalan** [apache/celeborn#2520] [Q&A]
In my testing most of the jobs are improved compared without celeborn But there is no improve on my most slow and complex query.

Any other information required for  it?
How to troubleshoot it ?

**Comment by FMX** [comment]
Hi @njalan . Can you share your cluster information and your workload information?
You'll have to find out the most time-consuming part of your query. If the shuffle-related process is the one, Celeborn will help a lot.

**Reply by njalan** [reply]
@FMX  Spark version is 3.0.1 and hadoop is 3.2.2 with 17  work nodes.
What is any other cluster information you need? 
How to get   workload information? what is the metrics for it?

**Comment by njalan** [comment]
@FMX How to find ind out the most time-consuming part of   query? Do you have any links for it?
